source/unified-test-format/unified-test-format.rst

```diff
@@ -1893,7 +1893,10 @@ test runner SHOULD use the client entity associated with the session
 to execute the ``configureFailPoint`` command. In this case, the test runner
 MUST also ensure that this command is excluded from the list of observed
 command monitoring events for this client (if applicable). If such an API is
-not available, test runners MUST create a new MongoClient that is directly
+not available, but the test runner creates an internal MongoClient for each
+mongos, the test runner MAY use the internal MongoClient corresponding to
```

Comment:
I wasn't sure whether this should be a SHOULD or a MAY. It could be a SHOULD just for the sake of lessening the number of created clients, but it doesn't make a functional difference I don't think. Any opinions?

Reply:
The `failPoint` operation states:

> (fail points are not configured using an internal MongoClient)

I don't recall the reason for that, since `configureFailPoint` commands are already ignored by default in `observeEvents`. I presume the original text here suggested creating a new MongoClient only because we never clarified how the internal MongoClient(s) dealt with multiple mongos nodes.

In that case, I think it makes sense to use SHOULD here. If drivers have an API to target specific servers, they probably won't have any reason to create one internal MongoClient for mongos in the first place.

Reply:
I changed this to a SHOULD. 

> I presume the original text here suggested creating a new MongoClient only because we never clarified how the internal MongoClient(s) dealt with multiple mongos nodes.

given this, is it worth keeping the "Otherwise, test runners MUST..." segment below?

Reply:
No harm in leaving it there, but it also seems like it won't really be relevant so I have no objections to deleting it. I defer to you since the Swift implementation is one of the target audiences of this section :)